crates/bevy_core_pipeline/src/prepass/mod.rs

```diff
@@ -0,0 +1,131 @@
+//! Run a prepass before the main pass to get the depth and/or normals texture.
+//! The depth prepass texture is then used by the main pass to reduce overdraw.
+//!
+//! To enable the prepass, you need to add a `PrepassSettings` component to a `Camera`.
+//! Both textures are available on the `PrepassTextures` component attached to each `Camera` with a `PrepassSettings`
+//!
+//! Currently only works for 3d
+
+pub mod node;
+
+use bevy_ecs::prelude::*;
+use bevy_render::{
+    render_phase::{CachedRenderPipelinePhaseItem, DrawFunctionId, EntityPhaseItem, PhaseItem},
+    render_resource::{CachedRenderPipelineId, Extent3d},
+    texture::CachedTexture,
+};
+use bevy_utils::FloatOrd;
+
+/// Add a `PrepassSettings` component to a `Camera` to perform a depth and/or normal prepass.
+/// These textures are useful for various screen-space effects and reducing overdraw in the main pass.
+///
+/// The textures are automatically added to the default mesh view bindings. You can also get the raw textures
+/// by querying the `ViewPrepassTextures` component on the camera with the `PrepassSettings`.
+#[derive(Clone, Component)]
+pub struct PrepassSettings {
+    /// If true then depth values will be copied to a separate texture available to the main pass.
+    /// This will help reduce overdraw in the main pass.
+    pub output_depth: bool,
+    /// If true then vertex world normals will be copied to a separate texture available to the main pass.
+    pub output_normals: bool,
+}
+
+impl Default for PrepassSettings {
+    fn default() -> Self {
+        Self {
+            output_depth: true,
+            output_normals: false,
+        }
+    }
+}
+
+/// Textures that are written to by the prepass. The texture only exists if the relevant option on `PrepassSettings` is true.
+#[derive(Component)]
+pub struct ViewPrepassTextures {
+    /// The depth texture generated by the prepass.
+    /// Exist only if `PrepassSettings::output_depth` is true
+    pub depth: Option<CachedTexture>,
+    /// The normals texture generated by the prepass.
+    /// Exist only if `PrepassSettings::output_normals` is true
+    pub normals: Option<CachedTexture>,
+    /// The size of the textures
+    pub size: Extent3d,
+}
+
+pub struct Opaque3dPrepass {
```

Comment:
Simple docs please :)

Reply:
To be fair, the equivalent draw commands from the main pass also don't have docs, but I do agree that basic docs should be done here and for the main pass.

Reply:
Yep :) I try to enforce this as I see it, to avoid stuff getting worse.